src/filter.ts

```diff
@@ -524,15 +532,40 @@ type ExpandL2<S extends string> = S extends L2Shortcuts
     : S;
 
 // === Define some helpers for when one property implies the existence of others
-type Twins<V extends string> = V extends KeyOf<Equivalents> ? Equivalents[V]
-    : V;
-type Equivalents = {
+
+// merges twins based on L1 with those based on L1 and L2
+type AddTwins<L1 extends string, L2 extends string> =
+    | TwinsFromL1<L1, L2>
+    | TwinsFromL2<L1, L2>;
+
+// yields twins based on a given L1 property
+type TwinsFromL1<L1 extends string, L2 extends string> = L1 extends
+    KeyOf<L1Equivalents> ? L1Equivalents[L1]
+    : L2;
```

Comment:
Cross-level twin implementation. Yields the correct fields on L2 for given keys on L1.